backend/de.metas.business/src/main/java/de/metas/order/compensationGroup/OrderGroupRepository.java

```diff
@@ -460,12 +456,12 @@ else if (groupIds.isEmpty() || groupIds.get(0) == null)
 		}
 	}
 
-	private Group createNewGroupFromOrderLines(
-			@Nullable final OrderId expectedOrderId,
-			@NonNull final List<I_C_OrderLine> existingRegularOrderLines,
-			@NonNull final GroupTemplate newGroupTemplate,
-			@Nullable final ConditionsId contractConditionsId)
+	private Group createNewGroupFromOrderLines(@NonNull final RetrieveOrCreateGroupRequest request,
+											   @NonNull final List<I_C_OrderLine> existingRegularOrderLines)
```

Comment:
why existingRegularOrderLines is not part of the request?

Reply:
The RetrieveOrCreateGroupRequest is what's being passed into OrderGroupRepository. The existingRegularOrderLines are looked up in OrderGroupRepository